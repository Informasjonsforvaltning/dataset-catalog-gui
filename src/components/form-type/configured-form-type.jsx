import { reduxForm } from 'redux-form';

import FormType from './form-type.component';
import { asyncValidateDatasetInvokePatch } from '../../entrypoints/main/router/datasets/pages/dataset-page/formsLib/asyncValidateDatasetInvokePatch';

const config = {
  form: 'type',
  asyncValidate: asyncValidateDatasetInvokePatch
};

export const ConfiguredFormType = reduxForm(config)(FormType);
