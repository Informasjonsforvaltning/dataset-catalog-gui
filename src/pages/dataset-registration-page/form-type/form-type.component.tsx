import React from 'react';
import { Field } from 'redux-form';
import _ from 'lodash';

import localization from '../../../services/localization';
import { Helptext } from '../../../components/helptext/helptext.component';
import CheckBoxFieldType from './field-checkbox-type/field-checkbox.component';
import { typeValues } from '../dataset-registration-page.logic';

interface Props {
  syncErrors: any;
  type: any;
  isReadOnly: boolean;
}

export const FormType = (props: Props) => {
  const { syncErrors, isReadOnly, type } = props;
  return (
    <form>
      <div className="form-group">
        <Helptext
          title={localization.schema.type.helptext.type}
          term="Dataset_type"
        />
        {!isReadOnly && <Field name="type" component={CheckBoxFieldType} />}
        {isReadOnly && <div className="pl-3">{typeValues(type.values)}</div>}
        {_.get(syncErrors, 'errorType') && (
          <div className="alert alert-danger mt-3">
            {_.get(syncErrors, 'errorType')}
          </div>
        )}
      </div>
    </form>
  );
};
