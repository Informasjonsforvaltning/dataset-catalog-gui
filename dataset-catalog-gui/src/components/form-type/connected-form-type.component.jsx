import { getFormSyncErrors } from 'redux-form';
import { connect } from 'react-redux';
import _ from 'lodash';

import { ConfiguredFormType } from './configured-form-type';

const mapStateToProps = (state, ownProps) => {
  const { datasetItem } = ownProps;
  return {
    initialValues: {
      type: _.get(datasetItem, 'type', '')
    },
    syncErrors: getFormSyncErrors('type')(state)
  };
};

export const ConnectedFormType = connect(mapStateToProps)(ConfiguredFormType);
