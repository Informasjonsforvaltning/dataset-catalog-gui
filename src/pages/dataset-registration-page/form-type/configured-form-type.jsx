import { reduxForm } from 'redux-form';

import { FormType } from './form-type.component';
import { asyncValidateDatasetInvokePatch } from '../formsLib/asyncValidateDatasetInvokePatch';

const config = {
  form: 'type',
  asyncValidate: asyncValidateDatasetInvokePatch
};

export const ConfiguredFormType = reduxForm(config)(FormType);
