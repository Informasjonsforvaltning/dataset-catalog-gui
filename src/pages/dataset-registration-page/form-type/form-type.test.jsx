import React from 'react';
import { shallow } from 'enzyme';
import { FormType } from './form-type.component';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {
    syncErrors: {
      errorType: true
    }
  };
  wrapper = shallow(<FormType {...defaultProps} />);
});

test('should render FormType, { renderLandingpage  correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
