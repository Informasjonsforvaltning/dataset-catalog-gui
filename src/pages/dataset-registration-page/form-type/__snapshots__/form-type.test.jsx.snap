// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormType, { renderLandingpage  correctly 1`] = `
<form>
  <div
    className="form-group"
  >
    <withStateHandlers(Component)
      term="Dataset_type"
      title="Type"
    />
    <Field
      component={[Function]}
      name="type"
    />
    <div
      className="alert alert-danger mt-3"
    >
      <Component />
    </div>
  </div>
</form>
`;
